### PR TITLE
New app creation process documentation update

### PR DESCRIPTION
**Pull request by Vladimir-Vdovic**
Completely remade `Getting Started`, as well as `Creating an Extension ` and `Creating Screens and Shortcuts` inside `My First Extension`. 
`My First Extension` now continues from where `Getting Started` ends, using the same extension.
Also included flow improvements.

Replaced and/or removed all obsolete @shoutem/cli commands, and adjusted documentation to reflect new practices. This is especially relevant in `Publishing your App`.

### CODE REVIEW

File: docs/extensions/my-first-extension/_posts/1970-01-03-InitializingExtension.md [hunk 1]
```diff
@@ -89,34 +77,24 @@ Brief property explanations:
...
 <p class="image">
 <img src='{{ site.url }}/img/my-first-extension/extension-tab-extension.png'/>
 </p>
 
-However, when you go to the `Screen` tab and click on `+` to add new screen, under `Custom` category you won't see your extension.
-
-<p class="image">
-<img src='{{ site.url }}/img/my-first-extension/add-content-no-extension.png'/>
-</p>
+Extensions are installed into specific apps, not all apps on your account. When running `shoutem install` inside a cloned app, the extension will be installed into that app, otherwise you will be prompted to select your app from a list.
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-03-InitializingExtension.md]: Maybe "When running ..." is not needed.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 2]
```diff
@@ -6,27 +6,20 @@ section: My first extension
...
 
-Since the app needs to know which screen to open the first for some extension, we need to create a ***shortcut*** when creating that screen. A shortcut is a link to the starting screen of an extension. It's the item in the app navigation which opens the starting screen when a user taps on it.
+Since the app needs to know which screen to open first for some extension, we need to create a ***shortcut*** when creating that screen. A shortcut is a link to the starting screen of an extension. It's the item in the Main Navigation which opens the starting screen when a user taps on it.
 
-The list of restaurants is going to be the first screen, so let's create it and give it a shortcut:
+When we created the List screen with a shortcut in [Getting Started]({{ site.url }}/docs/extensions/tutorials/getting-started) using:
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: Without "When". Just say "We already created ...". This is maybe too long and complex sentence.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 3]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
+We already did this in Getting Started, but let's elaborate on this. Since the app is managed through the Builder, we needed to `push` the extension to Shoutem and `install` it into our app so we can use and preview it in the Builder:
 
 ```ShellSession
 $ shoutem push
 Uploading `Restaurants` extension to Shoutem...
 Success!
-```
-
-Open the `Add screen` modal in the Builder and select the `Custom` category. You can now see your `Restaurants` starting screen (the shortcut).
+$ shoutem install
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: This is not needed. Extensions are already installed into the app in Getting Started. However, it would be also good to say that it was already installed, and we only need to install it once and it will get all the updates on the consequent `shoutem push` commands.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 4]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
-<p class="image">
-<img src='{{ site.url }}/img/my-first-extension/add-modal-shortcut.png'/>
-</p>
+Extension installed.
+See it in the builder: {{ site.shoutem.builderURL }}/app/{{ site.example.appId }}
+```
 
-> #### Note
-> In case you don't see it, refresh the page in browser.
+After installing it, we opened the app in the Builder and added the extension's screen to Main navigation. Installing new extensions and adding their shortcuts to the app requires you to rebuild your local clone:
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: Again, say that it's already installed.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 5]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
 
-Click on `Restaurants` to insert that shortcut into your app's navigation.
+```ShellSession
+$ shoutem reconfigure
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: The command is `shoutem configure`.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 6]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
+Let's preview your app again. We can preview it in the Builder, but it might take some time while Builder bundles the entire app again. Every time you change an extension, you'd have to _push_ it again and then the Builder would need to re-bundle the whole app to add the changes. It's much faster to use the **Shoutem Preview** app (available for [iOS]({{ site.shoutem.previewAppiOS }}) and [Android]({{ site.shoutem.previewAppAndroid }})) and the Shoutem CLI, which can bundle just the changes made in the extension instead of re-bundling the entire app, like the Builder would.
 
-```ShellSession
-$ shoutem link
-Extension successfully linked. Please, kill the packager before running the app.
-```
+You can also run your apps on local emulators the same way with regular React Native apps, we have a [tutorial]({{ site.url }}/docs/extensions/tutorials/setting-local-environment) that offers an in-depth explanation on all the ways you can develop locally.
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: the same way with regular React Native apps (`react-native run-ios` / `react-native run-android`), ... What do you say to add that?

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 7]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
 ```ShellSession
 $ shoutem run
-Select your app: Restaurants ({{ site.example.appId }})
-Creating the bundle for your app...
+Scanning folders for symlinks in /path/to/Restaurants_{{ site.example.appId }}/node_modules
 ...
 ```
 
-The CLI will print a QR code for you to scan using the Shoutem Preview app, if you don't have the app, you can get it using the link printed above the QR code. After it's installed, the Shoutem Preview app will automatically your app.
+Let's scan the QR code and see where we stopped in Getting Started.
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: Link Getting Started.

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 8]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
+
+<p class="image">
+<img src='{{ site.url }}/img/tutorials/getting-started/03-lets-eat.png'/>
+</p>
 
 > #### Note
-> In the documentation the preview you see is from the Builder, instead of a screenshot from the Shoutem Preview app. This way you'll see the state of the web interface as well. If you only change your app code, just shake your phone with the Shoutem Preview app on and tap the "Reload" button. If you [`link` your extension]({{ site.url }}/docs/extensions/tutorials/setting-local-environment), you won't need to do `shoutem push` every time you change the app code!
+> In the documentation the preview you see is from the Builder, instead of a screenshot from the Shoutem Preview app. This way you'll see the state of the web interface as well.
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: This is a great hack! 👍

File: docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md [hunk 9]
```diff
@@ -122,52 +81,71 @@ export const themes = extension.themes;
...
+
+Now let's make a quick change to the app code so you can see it change in real time on the Shoutem Preview app. Open your `restaurants` extension's `List.js` screen file and add another line of text:
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-04-CreatingShortcutAndScreen.md]: Great idea to show immediately how easy it is to see the change in extension!

File: docs/extensions/my-first-extension/_posts/1970-01-07-WorkingWithData.md [hunk 10]
```diff
@@ -6,11 +6,26 @@ section: My first extension
 ---
 
 # Working with Data
-<hr />
 
-Lets fetch data from the Shoutem Cloud storage to the app. First, remove the assets folder, we don't need it anymore. Create a  `reducer.js` file in the `app` folder. This file will contain a `reducer` defining the initial app state and how the state changes.
+Let's fetch data from the Shoutem Cloud storage to the extension. First, remove the `app/assets` folder, we don't need it anymore. Also remove the `getRestaurants()` function from `List.js`.
 
-Our [@shoutem/redux-io](https://github.com/shoutem/redux-io) package has `reducers` and `actions` that communicate with the Shoutem CMS. The `storage` reducer retrieves data (e.g. restaurants) into a dictionary, while `collection` stores data ID's in an array to persist its order.
+```JavaScript{2-4}
```
Comment (tenodi) [inline docs/extensions/my-first-extension/_posts/1970-01-07-WorkingWithData.md]: I don't think you need to say so specifically what to remove. It wouldn't bother if that function would stay in the extension. Like this, maybe it's just unnecessary step. Leave it on "Also remove the, ...".

File: docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md [hunk 11]
```diff
@@ -105,34 +105,25 @@ This is the manual process for publishing an iOS app to the App Store.
...
-To do this, use the Shoutem CLI command:
-
-```ShellSession
-$ shoutem build-ios
-  Select your app: Restaurants ({{ site.example.appId }})
-  ...
-```
-
-Your app's build file will be located in the current folder.
+To prepare your app for release, you will have to use `shoutem configure --release` in the cloned app's directory. After doing this short step, the build process is functionally identical to a [normal React Native app](https://medium.com/react-native-development/deploying-a-react-native-app-for-ios-pt-1-a79dfd15acb8).
```
Comment (tenodi) [inline docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md]: Good job for the reference.

File: docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md [hunk 12]
```diff
@@ -105,34 +105,25 @@ This is the manual process for publishing an iOS app to the App Store.
...
+You will need to make an iOS distribution certificate and matching distribution provisioning profile for your app. Here's a short video tutorial on how to do it:
 
 <iframe width="560" height="315" src="https://www.youtube.com/embed/Xh2nnjttOwo" frameborder="0" allowfullscreen></iframe>
 
 You can also use [Apple's guide](https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html#//apple_ref/doc/uid/TP40012582-CH31-SW1) where each step is explained in more detail.
 
 #### Resigning App
 
-Using the created certificates, you'll need to resign your un/signed build file (.ipa) that you got in the Build step. You can do it quickly wtih the Terminal by using Fastlane's [sigh](https://github.com/fastlane/fastlane/tree/master/sigh) tool.
+If you have an unsigned build (.ipa), you will need to resign it. You can do it quickly with the Terminal by using Fastlane's [sigh](https://github.com/fastlane/fastlane/tree/master/sigh) tool.
```
Comment (tenodi) [inline docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md]: I think it's not "if", but rather "once you have an unsigned build", right?

File: docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md [hunk 13]
```diff
@@ -224,25 +215,15 @@ This is a manual process of publishing an android app in the Google Play Store.
...
+To prepare your app for release, you will have to use `shoutem configure --release` in the cloned app's directory. After doing this short step, the build process is functionally identical to a [normal React Native app](https://facebook.github.io/react-native/docs/signed-apk-android.html).
 
 ### Preparing for Store
 
 Once you have the build ready, you can start preparing your app for store.
 
 #### App Signing and Zipalign
 
-You need to sign the build file (.apk) that you got from the Shoutem CLI. You need to sign your android apk with your own keystore. Here's the [official documentation](https://developer.android.com/studio/publish/app-signing.html) on how to do that.
+If you generated an unsigned build (.apk) you need to sign it with your own keystore. Here's the [official documentation](https://developer.android.com/studio/publish/app-signing.html) on how to do that.
```
Comment (tenodi) [inline docs/extensions/tutorials/_posts/1970-01-01-PublishYourApp.md]: I think it's not "if", but rather "once you generated an unsigned build", right?

**Comment by tenodi** [comment]
It's good, but please fix those smaller things still.

Also:
- Remove CSV from the files, we don't need it anymore: csv/getting-started-about.csv
- Remove images not being used any more:
  - img/my-first-extension/extension-hello-world.png
  - img/tutorials/getting-started/shoutem-run.png
  - img/tutorials/getting-started/snack-attack.jpg

File: docs/extensions/my-first-extension/_posts/1970-01-03-InitializingExtension.md [hunk 14]
```diff
@@ -6,46 +6,34 @@ section: My first extension
...
 Enter information about your extension. Press `return` to accept (default) values.
 Title: Restaurants
 Version: 0.0.1
-Description: List of restaurants
-
-Initializing extension:
-...
-
-Extension initialized!
+Description: A restaurants extension.
```
Comment (Vladimir-Vdovic) [inline docs/extensions/my-first-extension/_posts/1970-01-03-InitializingExtension.md]: https://github.com/shoutem/extension-examples/pull/5

File: docs/extensions/my-first-extension/_posts/1970-01-07-WorkingWithData.md (hunk 10 above)
Comment (Vladimir-Vdovic) [inline docs/extensions/my-first-extension/_posts/1970-01-07-WorkingWithData.md]: Users regularly leave this function in their code, which causes errors because the function requires a file from the `assets` folder which doesn't exist. I know this because 90% of the builds that fail with a "restaurants" extension fail because of it.

File: docs/extensions/tutorials/_posts/1970-01-01-GettingStarted.md [hunk 15]
```diff
@@ -21,47 +21,36 @@ Shoutem prepared and [open sourced](https://github.com/shoutem/extensions) a lot
...
-
-```ShellSession
-$ npm install -g @shoutem/cli
-```
-
-> #### Note
-> If previous command fails because of _permission_ issues, you need to run it with `sudo` permission: `sudo npm install -g @shoutem/cli`.
-
-Register your developer name. Use the `shoutem login` command with your Shoutem credentials ("{{ site.example.devName }}" is used as a developer name in this example).
+Start by using the `shoutem login` command with your Shoutem credentials ("{{ site.example.devName }}" is used as a developer name in this example).
```
Comment (Vladimir-Vdovic) [inline docs/extensions/tutorials/_posts/1970-01-01-GettingStarted.md]: Added this, not sure why the change wasn't covered. Probably because it was added in a different line.

**Comment by Vladimir-Vdovic** [comment]
@tenodi 
Addressed all requested changes. Can resolve conflicts when it's time to merge.
